streams/writable-streams/reentrant-strategy.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+// These tests exercise the pathological case of calling WritableStream* methods from within the strategy.size()
+// callback. This is not something any real code should ever do. Failures here indicate subtle deviations from the
+// standard that may affect real, non-pathological code.
+
+if (self.importScripts) {
+  self.importScripts('/resources/testharness.js');
+  self.importScripts('../resources/test-utils.js');
+  self.importScripts('../resources/recording-streams.js');
+}
+
+const error1 = { name: 'error1' };
+
+promise_test(() => {
+  let writer;
+  const strategy = {
+    size(chunk) {
+      if (chunk > 0) {
+        writer.write(chunk - 1);
+      }
+      return chunk;
+    }
+  };
+
+  const ws = recordingWritableStream({}, strategy);
+  writer = ws.getWriter();
+  return writer.write(2)
+      .then(() => Promise.resolve()) // See next test for why this is needed.
```

Comment:
could you please elaborate this comment a bit more? "next test" is 'writer.write() promises should resolve in the standard order'?

i guess this is to run micro tasks here to run the queued sink.write(), but not sure how it's explained in the next test.